Sources/Apodini/Modifier/GuardModifier.swift

```diff
@@ -33,11 +33,12 @@ struct GuardContextKey: ContextKey {
 
 
 public struct GuardModifier<C: Component>: Modifier {
-    public typealias Response = C.Response
+    public typealias ModifiedComponent = C
     
-    let component: C
+    public let component: C
     let `guard`: LazyGuard
     
+    public var content: some Component { EmptyComponent() }
     
     init<G: Guard>(_ component: C, guard: @escaping () -> G) {
         precondition(((try? typeInfo(of: G.self).kind) ?? .none) == .struct, "Guard \((try? typeInfo(of: G.self).name) ?? "unknown") must be a struct")
```

Comment:
Maybe we can reuse the `AssertTypeIsStruct` defined in `Component.swift` here as well. I think similar code is also found for ResponseTansformers and for RequestInjectibles (probably located in `Traversable.swift` as recently refactored by Max).